lib/middleware/request-handler.ts

```diff
@@ -390,92 +529,36 @@ export function onPollSubscriptions(req: Interface.IOurRequest,
     });
 }
 
-export function onUnsubscribe(req: Interface.IOurRequest,
-                              res: Interface.IOurResponse,
-                              next: restify.Next): void
-{
-    if (!req.apiSession.activeSubscriptions.size) {
-        req.log.debug('No active subscriptions.');
-        return next(new restify.BadRequestError('No active subscriptions.'));
-    }
-
-    var subscriptions: Subscription[] = [];
-    // If no correlation Id specified,
-    // the default behavior is to unsubscribe all active subscriptions
-    if (!req.body) {
-        subscriptions = req.apiSession.activeSubscriptions.values();
-    } else {
-        // If we do receive req.body object, first check if it is valid(empty list is INVALID)
-        if (!_.has(req.body, 'correlationIds') ||
-            !_.isArray(req.body.correlationIds) ||
-            !req.body.correlationIds.length) {
-            req.log.debug('Invalid unsubscribe data.');
-            return next(new restify.InvalidArgumentError('Invalid unsubscribe data.'));
-        }
-        // Next, validate all correlation Ids
-        // Will error if any invalid correlation Id received
-        var isAllValid = true;
-        _.uniq(req.body.correlationIds).forEach((cid: number): boolean => {
-            if (req.apiSession.activeSubscriptions.has(cid)) {
-                subscriptions.push(req.apiSession.activeSubscriptions.get(cid));
-            } else {
-                isAllValid = false;
-                req.log.debug('Invalid correlation Id ' + cid + ' received.');
-                return false;
-            }
-        });
-        if (!isAllValid) {
-            return next(new restify.InvalidArgumentError('Invalid correlation id.'));
-        }
-    }
-
-    try {
-        var result: Object[] = [];
-        req.blpSession.unsubscribe(subscriptions);
-        subscriptions.forEach((sub: Subscription): void => {
-            sub.removeAllListeners();
-            if (!sub.buffer.isEmpty()) {
-                result.push(sub.buffer.startNewBuffer());
-            }
-            req.apiSession.activeSubscriptions.delete(sub.correlationId);
-            req.apiSession.receivedSubscriptions.delete(sub.correlationId);
-        });
-        req.log.debug({ activeSubscriptions:
-                       req.apiSession.activeSubscriptions.size },
-                      'Unsubscribed.');
+var SUBSCRIPTION_ACTION_MAP: ActionMap = {
```

Comment:
Let's put global constants at the top of the file between imports and functions.